File: treatment/src/main/java/com/hartwig/actin/treatment/ctc/CTCDatabase.java
package com.hartwig.actin.treatment.ctc;

import java.util.List;
import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class CTCDatabase {

    @NotNull
    public abstract List<CTCDatabaseEntry> entries();

    @NotNull
    public abstract Set<String> studyMETCsToIgnore();

    @NotNull
    public abstract Set<Integer> unmappedCohortIds();

}

File: trial/src/main/kotlin/com/hartwig/actin/trial/ctc/config/CTCDatabase.kt
package com.hartwig.actin.trial.ctc.config

data class CTCDatabase(
    val entries: List<CTCDatabaseEntry>,
    val studyMETCsToIgnore: Set<String>,
    val unmappedCohortIds: Set<Int>,
    val mecStudiesNotInCTC: Set<String>
)